internal/httpserve/handlers/questionnaire.go

```diff
@@ -0,0 +1,196 @@
+package handlers
+
+import (
+	"time"
+
+	echo "github.com/theopenlane/echox"
+	"github.com/theopenlane/iam/auth"
+	"github.com/theopenlane/utils/contextx"
+
+	"github.com/theopenlane/core/internal/ent/generated"
+	"github.com/theopenlane/core/internal/ent/generated/assessment"
+	"github.com/theopenlane/core/internal/ent/generated/assessmentresponse"
+	"github.com/theopenlane/core/internal/ent/generated/privacy"
+	"github.com/theopenlane/core/pkg/enums"
+	models "github.com/theopenlane/core/pkg/openapi"
+)
+
+// GetQuestionnaire retrieves questionnaire template configuration for authenticated anonymous users
+func (h *Handler) GetQuestionnaire(ctx echo.Context, openapi *OpenAPIContext) error {
+	if isRegistrationContext(ctx) {
+		response := models.GetQuestionnaireResponse{
+			Jsonconfig: map[string]any{},
+		}
+		return h.Success(ctx, response, openapi)
+	}
+
+	reqCtx := ctx.Request().Context()
+
+	anonUser, ok := auth.AnonymousQuestionnaireUserFromContext(reqCtx)
+	if !ok {
+		return h.Unauthorized(ctx, ErrMissingQuestionnaireContext, openapi)
+	}
+
+	assessmentID := anonUser.AssessmentID
+	if assessmentID == "" {
+		return h.BadRequest(ctx, ErrMissingAssessmentID, openapi)
+	}
+
+	email := anonUser.SubjectEmail
+	if email == "" {
+		return h.BadRequest(ctx, ErrMissingEmail, openapi)
+	}
+
+	allowCtx := privacy.DecisionContext(reqCtx, privacy.Allow)
+	allowCtx = contextx.With(allowCtx, auth.QuestionnaireContextKey{})
+	allowCtx = auth.WithAnonymousQuestionnaireUser(allowCtx, anonUser)
+
+	assessmentResponseEntity, err := h.DBClient.AssessmentResponse.Query().
+		Where(
+			assessmentresponse.AssessmentIDEQ(assessmentID),
+			assessmentresponse.EmailEQ(email),
+		).
+		Only(allowCtx)
+	if err != nil && !generated.IsNotFound(err) {
+		return h.InternalServerError(ctx, err, openapi)
+	}
+
+	if assessmentResponseEntity != nil {
+		if assessmentResponseEntity.Status == enums.AssessmentResponseStatusCompleted {
+			return h.BadRequest(ctx, ErrAssessmentResponseAlreadyCompleted, openapi)
+		}
+
+		if !assessmentResponseEntity.DueDate.IsZero() && time.Now().After(assessmentResponseEntity.DueDate) {
+			_, err = h.DBClient.AssessmentResponse.UpdateOneID(assessmentResponseEntity.ID).
+				SetStartedAt(assessmentResponseEntity.StartedAt).
+				SetStatus(enums.AssessmentResponseStatusOverdue).
+				Save(allowCtx)
```

Comment:
believe you can use `Exec` here instead:
```suggestion
		 err = h.DBClient.AssessmentResponse.UpdateOneID(assessmentResponseEntity.ID).
				SetStartedAt(assessmentResponseEntity.StartedAt).
				SetStatus(enums.AssessmentResponseStatusOverdue).
				Exec(allowCtx)
```